rllib/env/env_runner_group.py

```diff
@@ -571,33 +580,43 @@ def sync_weights(
                     "`from_worker_or_trainer` is None. In this case, EnvRunnerGroup "
                     "should have local_env_runner. But local_env_runner is also None."
                 )
+            # New API stack: Use the unified `get_state` API.
             if self._remote_config.enable_env_runner_and_connector_v2:
-                weights = weights_src.get_state(
-                    components="rl_module",
-                    inference_only=inference_only,
-                    module_ids=policies,
+                modules = (
+                    [COMPONENT_RL_MODULE + "/" + p for p in policies]
+                    if policies is not None
+                    else [COMPONENT_RL_MODULE]
                 )
+                # LearnerGroup has-a Learner has-a RLModule.
                 if isinstance(weights_src, LearnerGroup):
-                    weights = weights["learner_state"]["rl_module"]
+                    rl_module_state = weights_src.get_state(
+                        components=[COMPONENT_LEARNER + "/" + m for m in modules],
+                        inference_only=inference_only,
```

Comment:
AH here it is. But this gets the state with the argument `inference_only` - why not leaving it in the `get_state` method of the `RLModule`s? Imo this makes it more clear to users what is happening.